docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_integrated_cloudwatch.mdx

```diff
@@ -23,6 +23,189 @@ CISAWS, CBP, HIPAA, SOC2, GDPR, NIST, AWSWAF, NISTCSF, PCIDSS, FedRAMP
+
+   You can use the `get-trail-status` command to get the status of a specific trail. Replace `<TrailName>` with the name of the trail you want to check.
+
+   Command:
+   ```
+   aws cloudtrail get-trail-status --name <TrailName>
+   ```
+
+3. **Check the CloudWatch Logs log group:**
+   If a trail has a CloudWatch Logs log group associated with it, you need to check if the log group is receiving log events from CloudTrail. This can be done by checking the `lastEventReceived` field in the log group's settings.
+
+   You can use the `describe-log-groups` command to get the settings of a specific log group. Replace `<LogGroupName>` with the name of the log group you want to check.
+
+   Command:
+   ```
+   aws logs describe-log-groups --log-group-name <LogGroupName>
+   ```
+
+4. **Check the CloudWatch Logs metric filters:**
+   Finally, you need to check if the log group has any metric filters that match CloudTrail events. This can be done by checking the `filterPattern` field in the metric filter's settings.
+
+   You can use the `describe-metric-filters` command to get the metric filters of a specific log group. Replace `<LogGroupName>` with the name of the log group you want to check.
+
+   Command:
+   ```
+   aws logs describe-metric-filters --log-group-name <LogGroupName>
+   ```
+   
+If any of these checks fail, it means that CloudTrail events are not being monitored by CloudWatch Logs.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Import necessary libraries and establish a session**: To start with, you need to import the necessary libraries in your Python script. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of AWS services like Amazon S3, Amazon EC2, etc. 
+
+```python
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+```
+Then, establish a session using your AWS credentials.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    aws_session_token='SESSION_TOKEN',
+)
+```
+
+2. **Create a CloudTrail client**: After establishing a session, create a CloudTrail client to interact with the service.
+
+```python
+cloudtrail = session.client('cloudtrail')
+```
+
+3. **List all trails and check for CloudWatch Logs integration**: Now, list all trails and check if they are integrated with CloudWatch Logs. If the "CloudWatchLogsLogGroupArn" attribute is not present, it means that the trail is not monitored by CloudWatch Logs.
+
+```python
+def check_trail_monitoring():
+    response = cloudtrail.describe_trails()
+    for trail in response['trailList']:
+        if 'CloudWatchLogsLogGroupArn' not in trail:
+            print(f"Trail {trail['Name']} is not monitored by CloudWatch Logs")
+```
+
+4. **Call the function**: Finally, call the function to check the monitoring status of all trails.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>